gradle/dependenciesMetadataRules.gradle.kts

```diff
@@ -0,0 +1,230 @@
+import org.gradle.api.artifacts.dsl.ComponentMetadataHandler
+
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+subprojects {
+    dependencies {
+        components {
+            // Gradle distribution - minify: remove unused transitive dependencies
+            withModule(library("maven3")) {
```

Comment:
Just a remark, but given the amount of duplication of `withModule(library("...")) { allVariants { withDependencies { ... }`, we could just introduce a method that does it and make the code more readable.

```
withLibraryDependencies("maven3") {
 ...
}
```

Reply:
:+1: It can always start as a local function.

Reply:
That's a good idea. Done.